internal/handlers/take_backup.go

```diff
@@ -37,90 +43,80 @@ func TBOperationHandler(
-			tb.GetId().String(),
-			tb.YdbOperationId,
-			err,
-		)
+	ydbOpResponse := LookupYdbOperationStatus(ctx, client, conn, operation, tb.YdbOperationId, tb.CreatedAt, config)
+	if ydbOpResponse.shouldAbortHandler {
+		operation.SetState(ydbOpResponse.opState)
+		operation.SetMessage(ydbOpResponse.opMessage)
+		return db.UpdateOperation(ctx, operation)
+	}
+	if ydbOpResponse.err != nil {
+		return ydbOpResponse.err
+	}
+	if ydbOpResponse.opResponse == nil {
+		return nil
+	}
+	opResponse := ydbOpResponse.opResponse
+
+	backupToWrite := types.Backup{
+		ID:     tb.BackupId,
+		Status: types.BackupStateUnknown,
 	}
 	switch tb.State {
 	case types.OperationStatePending:
 		{
-			if !opInfo.GetOperation().Ready {
-				//if pending: return op, nil
-				//if backup deadline failed: cancel operation. (skip for now)
-				return nil
-			}
-			if opInfo.GetOperation().Status == Ydb.StatusIds_SUCCESS {
-				//upsert into operations (id, status) values (id, done)?
-				//db.StartUpdate()
-				//.WithUpdateBackup()
-				//.WithYUpdateOperation()
-				err = db.UpdateBackup(ctx, tb.BackupId, types.BackupStateAvailable)
-				if err != nil {
-					return fmt.Errorf(
-						"error updating backup table, operation id %s: %w",
-						tb.GetId().String(),
-						err,
+			if !opResponse.GetOperation().Ready {
+				if DeadlineExceeded(tb.CreatedAt, config) {
+					err = CancelYdbOperation(ctx, client, conn, operation, tb.YdbOperationId, db, "TTL")
+					if err != nil {
+						return err
+					}
+					backupToWrite.Status = types.BackupStateError
+					return db.ExecuteUpsert(
+						ctx, getQueryBuilder().WithUpdateOperation(operation).WithUpdateBackup(backupToWrite),
 					)
+				} else {
+					return nil
 				}
+			} else if opResponse.GetOperation().Status == Ydb.StatusIds_SUCCESS {
+				backupToWrite.Status = types.BackupStateAvailable
 				operation.SetState(types.OperationStateDone)
 				operation.SetMessage("Success")
+			} else if opResponse.GetOperation().Status == Ydb.StatusIds_CANCELLED {
+				backupToWrite.Status = types.BackupStateError
+				operation.SetState(types.OperationStateError)
+				operation.SetMessage("got CANCELLED status for PENDING operation")
 			} else {
-				//op.State = Error
-				//upsert into operations (id, status, message) values (id, error, message)?
-				err = db.UpdateBackup(ctx, tb.BackupId, types.BackupStateError)
-				if err != nil {
-					return fmt.Errorf(
-						"error updating backup table, operation id %s: %w",
-						tb.GetId().String(),
-						err,
-					)
-				}
-				if opInfo.GetOperation().Status == Ydb.StatusIds_CANCELLED {
-					operation.SetMessage("got CANCELLED status for PENDING operation")
-				} else {
-					operation.SetMessage(types.IssuesToString(opInfo.GetOperation().Issues))
-				}
+				backupToWrite.Status = types.BackupStateError
 				operation.SetState(types.OperationStateError)
+				operation.SetMessage(types.IssuesToString(opResponse.GetOperation().Issues))
```

Comment:
why do not move IssuesToString to LookupYdbOperationResponse as method ?
opResponse.Issues() -> string ?